ports/catacomb3d/Catacomb 3D.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+XDG_DATA_HOME=${XDG_DATA_HOME:-$HOME/.local/share}
+
+if [ -d "/opt/system/Tools/PortMaster/" ]; then
+  controlfolder="/opt/system/Tools/PortMaster"
+elif [ -d "/opt/tools/PortMaster/" ]; then
+  controlfolder="/opt/tools/PortMaster"
+elif [ -d "$XDG_DATA_HOME/PortMaster/" ]; then
+  controlfolder="$XDG_DATA_HOME/PortMaster"
+else
+  controlfolder="/roms/ports/PortMaster"
+fi
+
+source $controlfolder/control.txt
+
+[ -f "${controlfolder}/mod_${CFW_NAME}.txt" ] && source "${controlfolder}/mod_${CFW_NAME}.txt"
+get_controls
+
+GAMEDIR="/$directory/ports/catacomb3d"
+
+export LD_LIBRARY_PATH="$GAMEDIR/libs.${DEVICE_ARCH}:$LD_LIBRARY_PATH"
+export SDL_GAMECONTROLLERCONFIG="$sdl_controllerconfig"
+
+$ESUDO chmod 777 -R $GAMEDIR/*
```

Comment:
That's brutal 😅

Reply:
Linux permissions, esp on these units, have left me scarred.  If I didn't do this, EDuke32 was one that failed to save games because even if the port created the folder, it wouldn't have permissions to write to that directory even if the files themselves got placed in them with full perms.  I never trust that the permissions won't cause issues.  I also keep this tied to just the game directory in question (certainly not going to do this outside of the gamedir like to system folders, etc. lol - THAT would be really bad).